### ISSUE TITLE
Question about ablation study reproduction

### ISSUE THREAD

**Issue by Hil-create** [Megum1/CO-SPY#11]
Hi, thanks for sharing this great work!
I’m currently trying to reproduce the results in your paper, especially the ablation study. I noticed that the repository provides scripts for training individual components (semantic/artifact), but I couldn’t find a dedicated setup or scripts for running the full ablation experiments as reported in the paper.
Would it be possible for you to share the code or configurations used for the ablation study?
Thanks a lot for your help!

**Comment by Megum1** [comment]
> Hi, thanks for sharing this great work! I’m currently trying to reproduce the results in your paper, especially the ablation study. I noticed that the repository provides scripts for training individual components (semantic/artifact), but I couldn’t find a dedicated setup or scripts for running the full ablation experiments as reported in the paper. Would it be possible for you to share the code or configurations used for the ablation study? Thanks a lot for your help!

Hi, thanks a lot for your interest in our project!

We actually do not provide a dedicated script specifically for running the ablation study... but the ablation run by training and evaluating each component separately, as well as the fused model.

You can follow the steps below:

**1. Train individual components**

```bash
# Train the semantic component (e.g., on CNNDet training set)
python main.py \
    --phase train \
    --gpu 0 \
    --mode branch \
    --branch semantic \
    --train_dataset progan \
    --epochs 10

# Train the artifact component
python main.py \
    --phase train \
    --gpu 1 \
    --mode branch \
    --branch artifact \
    --train_dataset progan \
    --epochs 20
```

**2. Train the fused (CO-SPY) model**

```bash
# Calibrate the combined CO-SPY detector
python main.py \
    --phase train \
    --gpu 2 \
    --mode fusion \
    --train_dataset progan \
    --epochs 2
```

**3. Evaluate each model**

```bash
python main.py --gpu 0 --phase eval --train_dataset progan --mode branch --branch semantic
python main.py --gpu 0 --phase eval --train_dataset progan --mode branch --branch artifact
python main.py --gpu 0 --phase eval --train_dataset progan --mode fusion
```

You can then compare the performance of these models to obtain the ablation results.

Hope this helps! Thanks again for your question.

**Comment by Hil-create** [comment]
Hi, thank you for your previous help!
I have another question regarding the end-to-end training.
I noticed that at epoch 0, the accuracy already reaches 100%. In this case, is it still necessary to continue training for more epochs? Or does this indicate something like overfitting, data leakage, or an issue in the training setup?
I would really appreciate your clarification.
Thanks in advance!

<img width="2018" height="528" alt="Image" src="https://github.com/user-attachments/assets/33bc975c-de38-403f-bc62-2ba74c650b18"

**Comment by Youkeen** [comment]
The performance of my own reproduce(trained in person) is far worse,and the average value is only 65.66. What about you?

**Comment by Megum1** [comment]
Hi Both,

Thank you for your questions!

In our experience, end-to-end training tends to overfit, mainly because the constraints are not strong enough while the learning space is too large... So if the accuracy reaches 100% at epoch 0, it usually suggests that the model is fitting the training objective too easily, rather than indicating that further end-to-end training would be beneficial..

This does not necessarily mean there is data leakage or an implementation issue, but it is still a good idea to double-check the data split and training setup to rule out those possibilities.

In general, we find that training the components separately and then fusing them leads to better performance than full end-to-end training.

I hope this helps clarify the issue.

**Comment by Youkeen** [comment]
I'm reaching out to inquire about the discrepancy in results. When directly applying the provided pre-trained weights, the model achieves an AP of 96.84 and an Acc of 87.86. However, when I attempt to replicate the results by first training each component separately and then fusing them as described, the AP drops to 90.91 and the Acc to 65.66.
I would greatly appreciate it if you could offer some insights or guidance on how to resolve this issue. Thank you very much for your time and assistance.

**Comment by Megum1** [comment]
> I'm reaching out to inquire about the discrepancy in results. When directly applying the provided pre-trained weights, the model achieves an AP of 96.84 and an Acc of 87.86. However, when I attempt to replicate the results by first training each component separately and then fusing them as described, the AP drops to 90.91 and the Acc to 65.66. I would greatly appreciate it if you could offer some insights or guidance on how to resolve this issue. Thank you very much for your time and assistance.

Thank you very much for your careful observation, and we sincerely apologize for the discrepancy.

The issue is likely caused by the training set we previously uploaded. As far as we remember, that version was downloaded and processed from a HuggingFace repository, possibly this one:

https://huggingface.co/datasets/shunk031/MSCOCO

There may have been some differences in the post-processing steps, which could lead to a gap between the reproduced results and the results obtained using our released pre-trained weights.

We have now updated the training set preparation script here:

https://github.com/Megum1/CO-SPY/blob/main/data/train/sd-v1_4/download.sh

Using the updated training data should produce results much closer to those of the provided pre-trained model.

We are very sorry for the confusion this has caused. Since some time has passed, we may not remember every detail of the original training parameters clearly, but we hope the updated dataset preparation script can help resolve the reproduction issue.

Thank you again for your patience and for pointing this out. Hope this could help!
